### PR TITLE
Add a hotkey to hard-wrap the current paragraph or the selection

### PR DESCRIPTION
**Pull request by inducer**
This makes use of the `wrap` add-on that already comes with CodeMirror.

### CODE REVIEW

File: public/js/lib/editor/index.js [hunk 1]
```diff
@@ -121,6 +121,30 @@ export default class Editor {
       },
       'Shift-Ctrl-Backspace': cm => {
         utils.wrapTextWith(this.editor, cm, 'Backspace')
+      },
+      'Ctrl-\\': cm => {
```
Comment (SuperSandro2000) [inline public/js/lib/editor/index.js:125]: Is this <kbd>Ctrl</kbd> + <kbd>\\</kbd> ? That will most likely no work on German keyboard cause we don't have <kbd>\\</kbd> as a primary key.

Comment (inducer) [reply public/js/lib/editor/index.js:125]: All the more reason to switch to a US layout. :) I know what I'm talking about, I grew up with the German layout. It's terrible.

On a more serious note, got any suggestions? I'm happy to switch this to something that's a bit more globally compatible.

Comment (SuperSandro2000) [reply public/js/lib/editor/index.js:125]: If you can convince my muscle memory to change. :) If you know all the keys by heart you won't change anytime soon if you not really like change.

What about <kbd>Shift</kbd> + <kbd>Enter</kbd>?

Comment (inducer) [reply public/js/lib/editor/index.js:125]: OK with me, but that's widely used (Word uses it for a non-paragraph-breaking line break). If my suggestion was too off-Broadway, maybe Shift-Enter is too on-Broadway? <kbd>Ctrl</kbd>+<kbd>.</kbd> maybe?

**Comment by inducer** [comment]
Does anyone know why Travis isn't completing?

**Comment by SuperSandro2000** [comment]
It is green but status is not reported back to github. https://travis-ci.com/github/codimd/server/builds/167981182

File: public/js/lib/editor/index.js (hunk 1 above)
Comment (SuperSandro2000) [reply public/js/lib/editor/index.js:125]: I am not sure about this.

**Comment by SISheogorath** [comment]
I may ask the usually obvious question: Why? Like what problem do we try to solve here? And have we checked that this doesn't cause other issues (i.e. formatting issues)?

**Comment by inducer** [comment]
> I may ask the usually obvious question: Why? Like what problem do we try to solve here?

I'll preface this by saying that I try to stick to 80 chars/line in any text document. I find long, wrapped lines difficult to navigate. Vertical cursor motion becomes more coarse-grained, while horizontal cursor motion remains inefficient. Plus indentation levels are less apparent. (Needless to say I'm a `breaks: false` person.)

This change helps me maintain that invariant in my documents without the need to do manual word wrapping.

Another reason to like short-ish lines: `diff` produces better output.

> And have we checked that this doesn't cause other issues (i.e. formatting issues)?

Not sure what you're looking for here. It's not perfect. It doesn't know about list items or blockqoute markers and will simply wrap them. But having this still beats re-wrapping a bunch of lines by hand.

**Comment by InnayTool** [comment]
@DerMolly and I consider to merge this into version 1.7. The CI says the commits by @kaushikcfd don't comply with the DCO, so we can't merge. Because of the rewrite in 2.0, this will be the last release were it can be merged it. Do you think it's apart from DCO ready to merge?

**Comment by InnayTool** [comment]
Thank you for your contribution! Since there was no further activity in this PR we decided to release 1.7rc1, which implies that this PR will not be merged. Unfortunately this means that there will be no release to merge it and I will close this PR.

**Pull request closed without merging**